test/perf/join-ig.js

```diff
@@ -0,0 +1,35 @@
+/* eslint-disable no-console, no-undef, new-cap */
+const puppeteer = require('puppeteer');
+const randomstring = require('randomstring');
+const microtime = require('microtime');
+
+module.exports = async () => {
+	console.log('starting performance test for joinAdInterestGroup');
+
+	const numInterestGroups = 100;
+
+	const browser = await puppeteer.launch();
+	const page = await browser.newPage();
+	await page.goto('http://localhost:3000/test/e2e/');
+	let totalExecutionTime = 0;
+
+	for (let i = 0; i < numInterestGroups; i++) {
+		const name = randomstring.generate(7);
```

Comment:
do you plan on using `randomString` in any way more than this?  If so, it's fine to leave it, but we could use the loop index to create a unique name and remove the lib altogether.

Reply:
it's a good point. I originally was thinking we could more effectively test performance by having a string that was more unique than just differing by one character, but I think I might be over-engineering it.

Let's leave it for now, but I would like to audit all dependencies before we go live and I think this might be a good candidate for removal.

Reply:
Yeah, good idea.  If we really want to get into creating "realistic" world examples of names and such, we should use `faker.js` then as a library choice.